### PR TITLE
Fix TOPPView 1d store bug and some warnings 

### LINKED ISSUE #6577: Newer Qt5 versions have many warnings
I needed to turn off Werror.
GHA, Ubuntu-latest

### PR DESCRIPTION
**Pull request by cbielow**
## Description

fixes #6577
and a bug which occurs when trying to store the visible data of a 1D spec/chrom.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms_gui/source/VISUAL/TOPPASEdge.cpp [hunk 1]
```diff
@@ -233,7 +233,7 @@ namespace OpenMS
       painter->translate(point);
       painter->rotate(text_angle);
       QFontMetrics fm(painter->fontMetrics());
-      int text_width = fm.width(str);
+      int text_width = fm.width(str); // replace with QFontMetrics::horizontalAdvance if Qt >= 5.11
```
Comment (jpfeuffer) [inline src/openms_gui/source/VISUAL/TOPPASEdge.cpp:236]: should we add a keyword for special todos that relate to minimum versions of contrib libs?
So we can easily search for them in the future, once in a while.

**Comment by cbielow** [comment]
Yes. Good idea.
Maybe
"CONTRIB_UPDATE_<LIB>_<MinVERSION>"

Am 01.02.2023 10:48 schrieb Julianus Pfeuffer ***@***.***>:

@jpfeuffer commented on this pull request.

________________________________

In src/openms_gui/source/VISUAL/TOPPASEdge.cpp<https://github.com/OpenMS/OpenMS/pull/6641#discussion_r1092987619>:

> @@ -233,7 +233,7 @@ namespace OpenMS
       painter->translate(point);
       painter->rotate(text_angle);
       QFontMetrics fm(painter->fontMetrics());
-      int text_width = fm.width(str);
+      int text_width = fm.width(str); // replace with QFontMetrics::horizontalAdvance if Qt >= 5.11


should we add a keyword for special todos that relate to minimum versions of contrib libs?
So we can easily search for them in the future, once in a while.

**Comment by jpfeuffer** [comment]
Yep, fine with me.

**Comment by cbielow** [comment]
should be good to go

**Comment by timosachsenberg** [comment]
only sirius stuff failed

**Pull request merged**